lib/libutee/include/remoteproc_pta.h

```diff
@@ -0,0 +1,150 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2023, STMicroelectronics
+ */
+
+#ifndef __REMOTEPROC_PTA_H
+#define __REMOTEPROC_PTA_H
+
+#include <stdint.h>
+#include <util.h>
+
+/*
+ * Interface to the pseudo TA which provides platform implementation
+ * of the remote processor management
+ */
+
+#define PTA_RPROC_UUID { 0x54af4a68, 0x19be, 0x40d7, \
+		{ 0xbb, 0xe6, 0x89, 0x50, 0x35, 0x0a, 0x87, 0x44 } }
+
+/* Hardware capability: firmware format */
+#define PTA_RPROC_HWCAP_FMT_ELF	BIT32(0)
+
+/* Hardware capability: image protection method */
+/* The platform supports load of segment with hash protection */
+#define PTA_RPROC_HWCAP_PROT_HASH_TABLE	BIT32(0)
+
+/**
+ * struct rproc_pta_key_info - public key information
+ * @algo:	Algorithm, defined by public key algorithms TEE_ALG_*
+ * @info_size:	Byte size of @info
+ * @info:	Append key information data
+ */
+struct rproc_pta_key_info {
+	uint32_t algo;
+	uint32_t info_size;
+	uint8_t info[];
+};
+
+static inline size_t rproc_pta_keyinfo_size(struct rproc_pta_key_info *keyinf)
+{
+	size_t s = 0;
+
+	if (!keyinf || ADD_OVERFLOW(sizeof(*keyinf), keyinf->info_size, &s))
+		return 0;
+
+	return s;
+}
+
+/*
+ * Platform capabilities.
+ *
+ * Get Platform firmware loader service capabilities.
+ *
+ * [in]  params[0].value.a:	Unique 32bit firmware identifier
```

Comment:
Actually, I wonder if `Unique 32bit remote processor identifier` is better.
The TA/PTA don't really care which system is implemented in the binary, rather where it's expected to be loaded, no?